### PR TITLE
Adds 3 mutant dog types + their puppies

### PR DESCRIPTION
**Pull request by Daved27hundred**
#### Summary
Content "Added mutated dogs as a parallel to the mutated cats we have ingame"

#### Purpose of change

I really like the mutant cats from #45065 and #59104, they're some nice flavor and their descriptions are creative. I thought dogs could use some bizarre mutant variants as well.

We now have the Translucent dog, the Triclopean dog, and the Unknowable dog. They should spawn like any other dog would, only rarer.
#### Describe the solution

Wrote some new dogs in mutant_mammals.json and put them in a monster group.

#### Describe alternatives you've considered

Maybe more "realistic" mutations?

#### Additional context

Hi Aerin

### CODE REVIEW

File: data/json/monsters/mutant_mammal.json [hunk 1]
```diff
@@ -427,5 +427,195 @@
...
+      "SMELLS",
+      "SWARMS",
+      "WARM"
+    ]
+  },
+  {
+    "id": "mon_dog_incorporeal_pup",
+    "type": "MONSTER",
+    "copy-from": "mon_dog_auscattle_pup",
+    "name": { "str": "translucent puppy" },
```
Comment (TheShadowFerret) [inline data/json/monsters/mutant_mammal.json:491]:
```suggestion
    "name": { "str": "translucent puppy", "str_pl": "translucent puppies" },
```

File: data/json/monsters/mutant_mammal.json [hunk 2]
```diff
@@ -427,5 +427,195 @@
...
+      "SMELLS",
+      "SWARMS",
+      "WARM"
+    ]
+  },
+  {
+    "id": "mon_dog_telekenetic_pup",
+    "type": "MONSTER",
+    "copy-from": "mon_dog_auscattle_pup",
+    "name": { "str": "triclopean puppy" },
```
Comment (TheShadowFerret) [inline data/json/monsters/mutant_mammal.json:554]:
```suggestion
    "name": { "str": "triclopean puppy", "str_pl": "triclopean puppies" },
```

File: data/json/monsters/mutant_mammal.json [hunk 3]
```diff
@@ -427,5 +427,195 @@
...
+      "SMELLS",
+      "SWARMS",
+      "WARM"
+    ]
+  },
+  {
+    "id": "mon_dog_antimemetic_pup",
+    "type": "MONSTER",
+    "copy-from": "mon_dog_auscattle_pup",
+    "name": { "str": "unknowable puppy" },
```
Comment (TheShadowFerret) [inline data/json/monsters/mutant_mammal.json:617]:
```suggestion
    "name": { "str": "unknowable puppy", "str_pl": "unknowable puppies" },
```

File: data/json/monsters/mutant_mammal.json [hunk 4]
```diff
@@ -427,5 +427,195 @@
...
+    "description": "Try as you might, your brain is unable to recall any details on what this puppy looks like. All you can remember is that it is a puppy, that it exists, and that it is cute.",
```
Comment (TheShadowFerret) [inline data/json/monsters/mutant_mammal.json:618]:
```suggestion
    "description": "Try as you might, your brain is unable to recall any details on what this puppy looks like.  All you can remember is that it is a puppy, that it exists, and that it is cute.",
```

**Review by TheShadowFerret — CHANGES_REQUESTED** [review]

File: data/json/monsters/mutant_mammal.json (hunk 4 above)
Comment (TheShadowFerret) [reply data/json/monsters/mutant_mammal.json:618]: What texture would an unknowable dog even have 🤔

Comment (Daved27hundred) [reply data/json/monsters/mutant_mammal.json:618]: grey silhouette of a dog, perhaps?

**Review by Maleclypse — APPROVED** [review]

**Pull request merged**